File: examples/Delphi/ExDlgU.pas
{$J+} {Writable constants}

unit ExDlgU;

interface

uses
  Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  StdCtrls, Db, DBTables,
  OvcBase, OvcCal, OvcCalc, OvcDlg, 
  OvcMoDg, OvcDbADg, OvcSplDg, OvcClkDg,
  OvcClcDg, OvcDbMDg, OvcCalDg;

type
  TForm2 = class(TForm)
    OvcCalendarDialog1: TOvcCalendarDialog;
    OvcDbMemoDialog1: TOvcDbMemoDialog;
    Button1: TButton;
    OvcCalculatorDialog1: TOvcCalculatorDialog;
    Button2: TButton;
    OrController1: TOvcController;
    Button3: TButton;
    OvcClockDialog1: TOvcClockDialog;
    DataSource1: TDataSource;
    Table1: TTable;
    Button4: TButton;
    Button5: TButton;
    OvcMemoDialog1: TOvcMemoDialog;
    Label1: TLabel;
    Memo1: TMemo;
    Button6: TButton;
    OvcDbAliasDialog1: TOvcDbAliasDialog;
    Button7: TButton;
    OvcSplashDialog1: TOvcSplashDialog;
    procedure Button1Click(Sender: TObject);
    procedure Button2Click(Sender: TObject);
    procedure Button3Click(Sender: TObject);
    procedure Button4Click(Sender: TObject);
    procedure OvcDbMemoDialog1HelpClick(Sender: TObject);
    procedure Button5Click(Sender: TObject);
    procedure FormCreate(Sender: TObject);
    procedure Button6Click(Sender: TObject);
    procedure Button7Click(Sender: TObject);
    procedure OvcSplashDialog1Click(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
    SelectedDate : TDateTime;
  end;

var
  Form2: TForm2;

implementation

{$R *.DFM}

procedure TForm2.FormCreate(Sender: TObject);
begin
  SelectedDate := Now;
end;

procedure TForm2.Button1Click(Sender: TObject);
begin
  {initialize calendar date}
  OvcCalendarDialog1.Calendar.Date := Now;
  {show dialog and retrieve date if OK pressed}
  if OvcCalendarDialog1.Execute then begin
    SelectedDate := OvcCalendarDialog1.Calendar.Date;
    Button1.Caption := DateToStr(OvcCalendarDialog1.Calendar.Date);
  end;
end;

procedure TForm2.Button2Click(Sender: TObject);
begin
  if OvcCalculatorDialog1.Execute then
    Button2.Caption := FloatToStr(OvcCalculatorDialog1.Calculator.DisplayValue);
end;

procedure TForm2.Button3Click(Sender: TObject);
begin
  OvcClockDialog1.Clock.DrawMarks := False;
  OvcClockDialog1.Execute;
end;

procedure TForm2.Button4Click(Sender: TObject);
begin
  if OvcDbMemoDialog1.Execute then begin
    {memo field in database has been updated}
  end;
end;

procedure TForm2.Button5Click(Sender: TObject);
begin
  OvcMemoDialog1.Lines.Assign(Memo1.Lines);
  if OvcMemoDialog1.Execute then begin
    {transfer memo data}
    Memo1.Lines.Assign(OvcMemoDialog1.Lines);
  end;
end;

procedure TForm2.Button6Click(Sender: TObject);
begin
  {initialize the stating alias} 
  OvcDbAliasDialog1.AliasName := 'DBDEMOS';
  if OvcDbAliasDialog1.Execute then begin
    {do something with alias, table, path}
    Button6.Caption := OvcDbAliasDialog1.TableName;
  end;
end;

procedure TForm2.OvcDbMemoDialog1HelpClick(Sender: TObject);
begin
  ShowMessage('Help button was clicked');
end;


procedure TForm2.Button7Click(Sender: TObject);
begin
  if not OvcSplashDialog1.Visible then
    OvcSplashDialog1.Show
  else
    OvcSplashDialog1.Close;
end;

procedure TForm2.OvcSplashDialog1Click(Sender: TObject);
begin
  OvcSplashDialog1.Close;
end;

end.

File: examples/CBuildr3/ExDlgu.cpp
//---------------------------------------------------------------------------
#include <vcl\vcl.h>
#pragma hdrstop

#include "ExDlgu.h"
//---------------------------------------------------------------------------
#pragma link "OvcCalDg"
#pragma link "OvcDlg"
#pragma link "OvcBase"
#pragma link "OvcDbMDg"
#pragma link "OvcClcDg"
#pragma link "OvcClkDg"
#pragma link "OvcMoDg"
#pragma link "OvcDbADg"
#pragma link "OvcSplDg"
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
	: TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TForm1::FormCreate(TObject *Sender)
{
  SelectedDate = Now();
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Button1Click(TObject *Sender)
{
  OvcCalendarDialog1->Calendar->Date = Now();
  if (OvcCalendarDialog1->Execute()) {
    SelectedDate = OvcCalendarDialog1->Calendar->Date;
    Button1->Caption = OvcCalendarDialog1->Calendar->Date;
  }
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Button3Click(TObject *Sender)
{
  OvcClockDialog1->Clock->DrawMarks = False;
  OvcClockDialog1->Execute();
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Button6Click(TObject *Sender)
{
  OvcDbAliasDialog1->AliasName = "BCDEMOS";
  if (OvcDbAliasDialog1->Execute())
    Button6->Caption = OvcDbAliasDialog1->TableName;
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Button7Click(TObject *Sender)
{
  if (!OvcSplashDialog1->Visible)
    OvcSplashDialog1->Show();
  else
    OvcSplashDialog1->Close();
}
//---------------------------------------------------------------------------
void __fastcall TForm1::OvcSplashDialog1Click(TObject *Sender)
{
  OvcSplashDialog1->Close();
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Button2Click(TObject *Sender)
{
  if( OvcCalculatorDialog1->Execute())
    Button2->Caption = String((double)OvcCalculatorDialog1->Calculator->DisplayValue);
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Button5Click(TObject *Sender)
{
  OvcMemoDialog1->Lines->Assign(Memo1->Lines);
  if (OvcMemoDialog1->Execute())
    Memo1->Lines->Assign(OvcMemoDialog1->Lines);
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Button4Click(TObject *Sender)
{
  if (OvcDbMemoDialog1->Execute()) {
//    {memo field in database has been updated}
  }
}
//---------------------------------------------------------------------------
void __fastcall TForm1::OvcDbMemoDialog1HelpClick(TObject *Sender)
{
  ShowMessage("Help button was clicked");
}
//---------------------------------------------------------------------------
